runtime/gc_check/CheckEngine.cpp

```diff
@@ -461,9 +461,9 @@ GC_CheckEngine::checkJ9Object(J9JavaVM *javaVM, J9Object* objectPtr, J9MM_Iterat
 	}
 
 #if defined(J9VM_ENV_DATA64)
-	if (OMR_ARE_ANY_BITS_SET(_cycle->getMiscFlags(), J9MODRON_GCCHK_VALID_INDEXABLE_DATA_ADDRESS) && extensions->objectModel.isIndexable(objectPtr)) {
+	if (extensions->isVLHGC() && OMR_ARE_ANY_BITS_SET(_cycle->getMiscFlags(), J9MODRON_GCCHK_VALID_INDEXABLE_DATA_ADDRESS) && extensions->objectModel.isIndexable(objectPtr) && javaVM->isIndexableDataAddrPresent) {
```

Comment:
why do we need this isVLHGC (here, and in Java/DDR part too) - would it not be sufficient just do check  isIndexableDataAddrPresent?